src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+        contactsAddrs.addrs[0].MailingStreet__c = 'New Street';
+        contactsAddrs.addrs[0].MailingCity__c = 'New City';
+        contactsAddrs.addrs[1].MailingStreet__c = 'New Street';
+        contactsAddrs.addrs[1].MailingCity__c = 'New City';
+
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.addrs;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            // verify that the Contact address is override
+            if (contact.is_Address_Override__c) { 
+                system.assert(!contact.MailingStreet.contains('street override'));
+                system.assert(!contact.MailingCity.contains('city override'));
+                system.assertEquals(contact.Current_Address__c, addrNew.Id);            
+            // verify that the Contact address is new
+            } else { 
+                system.assert(contact.MailingStreet.contains('New Street'));
+                system.assert(contact.MailingCity.contains('New City'));
+                system.assertNotEquals(null, contact.Current_Address__c);
+            }
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Insert new default addresses to Contact w/ existing default addresses  
+    verify:
+        Contact address matches new default
+        old default addresses no longer marked default
+    **********************************************************************************************************/            
+    @isTest
+    public static void insertNewDefaultAddr() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+        
+        // create additional addresses
+        List<Address__c> addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            addrs[i].Parent_Contact__c = contactsAddrs.contacts[i].Id;
+            addrs[i].Default_Address__c = true;
+            addrs[i].MailingStreet__c = 'New Default Street' + i;
+            addrs[i].MailingCity__c = 'New Default City' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;        
+        Test.stopTest();
+
+        // verify that the Contact addresses are new
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.